code/modules/power/engines/supermatter/supermatter.dm

```diff
@@ -908,12 +911,35 @@
 			l_power = 3,
 			l_color = SUPERMATTER_TESLA_COLOUR,
 		)
-	if(combined_gas > MOLE_PENALTY_THRESHOLD)
+	if(combined_gas > MOLE_PENALTY_THRESHOLD && get_integrity() > SUPERMATTER_DANGER_PERCENT)
 		set_light(
-			l_range = 4 + clamp(damage / 2, 10, 50),
+			l_range = 4 + clamp((450 - damage) / 10, 1, 50),
 			l_power = 3,
 			l_color = SUPERMATTER_SINGULARITY_LIGHT_COLOUR,
 		)
+	if(combined_gas > MOLE_PENALTY_THRESHOLD && get_integrity() < SUPERMATTER_DANGER_PERCENT)
+		var/darkness_strength = clamp((damage - 450) / 75, 1, 8) / 2
+		var/darkness_aoe = clamp((damage - 450) / 25, 1, 25)
+		set_light(
+			l_range = 4 + darkness_aoe,
+			l_power = -1 - darkness_strength,
+			l_color = "#ddd6cf")
+		if(!length(darkness_effects) && istype(src, /obj/machinery/atmospherics/supermatter_crystal/engine)) //Don't do this on movable sms oh god. Ideally don't do this at all, but hey, that's lightning for you
+			darkness_effects += new /obj/effect/abstract(locate(x-3,y+3,z))
```

Comment:
These numbers look a bit magic. I know containment is usually pretty standard, but you'd probably be better served iterating through and finding suitable diagonal spaces past any walls, just in case.

Reply:
I'll try doing a check at these locations for valid tiles